frontend/app/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -0,0 +1,82 @@
+import { BASE_POLLING_INTERVAL_MS } from "@/consts";
+import { Linterlint } from "@/domain/LinterLint";
+import { usePageVisibility } from "@/lib/usePageVisibility";
+import { getLinterLintsV1 } from "@/services/rest/RestService";
+import { Editor, useMonaco } from "@monaco-editor/react";
+import { useEffect, useRef } from "react";
+
+interface CodeEditorProps {
+	setProgram: (s: string) => void;
+}
+
+export default function CodeEditor({ setProgram }: Readonly<CodeEditorProps>) {
+	const monaco = useMonaco();
+	const linterOwner = "redCodeLinter";
+	const isPageVisible = usePageVisibility();
+	const timerIdRef = useRef<ReturnType<typeof setInterval> | null>(null);
+
+	useEffect(() => {
+		const pollingCallback = async () => {
+			console.debug("Polling linter warnings...");
+
+			const responseLints = await getLinterLintsV1(
+				monaco?.editor.getModels()[0].getValue() ?? "",
+			);
+			setLinterLinting(responseLints);
+		};
```

Comment:
should we explicity check if monaco is null here? since useMonaco is async, it could be possible for a race condition to occur here I think

Reply:
there is a peculiarity with the useMonaco hook that is explained in the docs: it updates two times, the first time when the page is loaded is it loaded with null and will contain the actual editor once it is fully loaded, so null checks are necessary yes.

Reply:
i also don't understand what kind of race condition you are talking about ? :sweat: